Add support for Vitest 3.1.x

boimw opened the pull request:
Fixes #8743.

This MR adds support for Vitest 3.1.x and it fixes peer dependency with `vitest`, `@vitest/coverage-istanbul` and similar packages.

---

<!--
Please don't delete the checkboxes <3
The following selections do not need to be completed if this PR only contains changes to .md files
-->

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: dependency update
- Wrangler / Vite E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because:
- Public documentation
  - [ ] TODO (before merge)
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/21700
  - [ ] Documentation not necessary because:
- Wrangler V3 Backport
  - [ ] TODO (before merge)
  - [ ] Wrangler PR: <!--e.g. <https://github.com/cloudflare/workers-sdk/pull/>...-->
  - [x] Not necessary because: not a Wrangler change.

<!--
Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?
In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.
-->

boimw commented:
Hey @petebacondarwin 👋  could you please review this dependency update? 🙇‍♂️

petebacondarwin commented:
Running the tests but this PR will need a bit more: a changeset and possible updates to the version of Vitest we are using in our testing and fixtures...

boimw commented:
> Running the tests but this PR will need a bit more: a changeset and possible updates to the version of Vitest we are using in our testing and fixtures...

Thanks @petebacondarwin 

I've added `changeset` and updated `vitest` versions inside `pnpm-workspace.yaml`. Could you please take another look?

penalosa commented on pnpm-workspace.yaml:
```diff
@@ -7,17 +7,17 @@ packages:
+  vitest: "~3.1.1"
   vite: "^5.4.14"
   "ws": "8.18.0"
-  esbuild: "0.24.2"
```
This PR shouldn't be updating esbuild—it should be limited to Vitest

penalosa reviewed (dismissed):
This looks generally good, but shouldn't update esbuild

boimw commented:
Updated, thanks for the support @petebacondarwin 

Back to you 🏓

petebacondarwin commented:
Looks like Vitest 3.1 is formatting snapshots slightly differently?

https://github.com/cloudflare/workers-sdk/actions/runs/14331654155/job/40169757071?pr=8804#step:5:1161

I'll update and push a commit.

boimw commented:
@petebacondarwin I just solved the conflicts, back to you 🏓

alsuren reviewed (approved):
I assume D1 has been tagged for a code owner review because of packages/wrangler/src/__tests__/d1/insights.test.ts , so I'm happy to approve that bit to unblock you (assuming that tests still pass), but please wait for someone on the wrangler team to approve the rest before merging.
